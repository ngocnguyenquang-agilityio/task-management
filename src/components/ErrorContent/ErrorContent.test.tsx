import { render, screen, fireEvent } from '@testing-library/react';

// Component
import { ErrorContent } from '.';

const mockProps = {
  onClick: jest.fn(),
};

describe('ErrorContent component', () => {
  test('Render component correctly', () => {
    expect(render(<ErrorContent {...mockProps} />)).toMatchSnapshot();
  });

  test('Show correct default title', () => {
    render(<ErrorContent {...mockProps} />);
    const title = screen.getByText('Something went wrong');

    expect(title).toBeInTheDocument();
  });

  test('Show correct default description', () => {
    render(<ErrorContent {...mockProps} />);
    const description = screen.getByText(
      'The page you were trying to reach is currently unavailable.',
    );

    expect(description).toBeInTheDocument();
  });

  test('Should render button element', () => {
    render(<ErrorContent {...mockProps} />);
    const buttonElement = screen.getByRole('button');

    expect(buttonElement).toBeInTheDocument();
  });

  test('Should call onSelect function', async () => {
    render(<ErrorContent {...mockProps} />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(mockProps.onClick).toHaveBeenCalledTimes(1);
  });

  test('Show correct title passing thourgh props', () => {
    const { getByText } = render(
      <ErrorContent title="Opp! Something went wrong" {...mockProps} />,
    );
    const title = getByText('Opp! Something went wrong');

    expect(title).toBeInTheDocument();
  });

  test('Show correct description passing thourgh props', () => {
    const { getByText } = render(
      <ErrorContent description="The page is unavailable." {...mockProps} />,
    );
    const description = getByText('The page is unavailable.');

    expect(description).toBeInTheDocument();
  });

  test('Should render label button passing thourgh props', () => {
    const { getByText } = render(
      <ErrorContent labelButton="Try again" {...mockProps} />,
    );
    const buttonElement = getByText('Try again');

    expect(buttonElement).toBeInTheDocument();
  });
});
