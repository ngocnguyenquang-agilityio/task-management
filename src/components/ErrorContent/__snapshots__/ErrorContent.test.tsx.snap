// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorContent component Render component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto flex flex-col items-center justify-center mt-[177px] sm:mt-14 lg:mt-52 mb-7.5"
      >
        <svg
          class="w-40 h-40 text-red-500 md:h-80 md:w-80"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.953 2C6.465 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.493 2 11.953 2zM13 17h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
        <div
          class="flex flex-col items-center gap-3 md:gap-5"
        >
          <h1
            class="text-center font-bold text-2xl md:text-3xl"
          >
            Something went wrong
          </h1>
          <h2
            class="text-center text-md md:text-xl"
          >
            The page you were trying to reach is currently unavailable.
          </h2>
          <button
            class="flex items-center rounded-lg p-2 disabled:cursor-not-allowed bg-neutral-800 text-white hover:bg-neutral-700 dark:bg-black dark:hover:opacity-70 disabled:bg-neutral-800/50 w-[88px] mt-6 md:mt-2"
            type="button"
          >
            Try again
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto flex flex-col items-center justify-center mt-[177px] sm:mt-14 lg:mt-52 mb-7.5"
    >
      <svg
        class="w-40 h-40 text-red-500 md:h-80 md:w-80"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.953 2C6.465 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.493 2 11.953 2zM13 17h-2v-2h2v2zm0-4h-2V7h2v6z"
        />
      </svg>
      <div
        class="flex flex-col items-center gap-3 md:gap-5"
      >
        <h1
          class="text-center font-bold text-2xl md:text-3xl"
        >
          Something went wrong
        </h1>
        <h2
          class="text-center text-md md:text-xl"
        >
          The page you were trying to reach is currently unavailable.
        </h2>
        <button
          class="flex items-center rounded-lg p-2 disabled:cursor-not-allowed bg-neutral-800 text-white hover:bg-neutral-700 dark:bg-black dark:hover:opacity-70 disabled:bg-neutral-800/50 w-[88px] mt-6 md:mt-2"
          type="button"
        >
          Try again
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
